subsys/bluetooth/host/l2cap.c

```diff
@@ -2838,12 +2776,7 @@ void bt_l2cap_init(void)
 	}
 
 #if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
-	k_fifo_init(&free_l2cap_tx_meta_data);
-	for (size_t i = 0; i < ARRAY_SIZE(l2cap_tx_meta_data_storage); i++) {
-		(void)memset(&l2cap_tx_meta_data_storage[i], 0,
-					sizeof(l2cap_tx_meta_data_storage[i]));
-		k_fifo_put(&free_l2cap_tx_meta_data, &l2cap_tx_meta_data_storage[i]);
-	}
+	sys_slist_init(&servers);
```

Comment:
This did not fail before because `servers` is static so it was initialized to an empty list.

Was a `bt_enable`/`bt_disable` loop broken if a server was registered?

Reply:
not sure, but I saw it fail on a bsim test. don't remember which one now, but it doesn't fail anymore.
anyway, the API still states you have to call init, regardless of `static`